### PR TITLE
ci: add test workflow

### PR DESCRIPTION
**Pull request by naorpeled**
- added a test workflow to make sure future breaking changes will be found before being merged

### CODE REVIEW

**Comment by naorpeled** [comment]
For some reason the tests are failing in Node 10 and 12,
not sure why but unfortunately won't be able to investigate further today.

**Comment by BenBaryoPX** [comment]
I believe the earliest node version supported is 14

**Comment by BenBaryoPX** [comment]
I've added some tests yesterday following another PR. Could you run the flow and see that everything works?

**Comment by naorpeled** [comment]
> I've added some tests yesterday following another PR. Could you run the flow and see that everything works?

Sure!

**Comment by naorpeled** [comment]
> I've added some tests yesterday following another PR. Could you run the flow and see that everything works?

The workflow is passing

**Review by BenBaryoPX — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "description": "Flatten JS AST",
   "main": "src/index.js",
   "scripts": {
+    "lint": "eslint .",
```
[the remark this answers is not in the document]
Comment (naorpeled) [reply package.json:7]: It's a dev dependency, therefore it would be available within npm install/ci context, so it's redundant.

**Review by BenBaryoPX — APPROVED** [review]

**Pull request merged**